### ISSUE TITLE
[Go] Conform to Golang idioms and best practices

### ISSUE THREAD

**Issue by neilotoole** [swagger-api/swagger-codegen#3518]
##### Description

There are several locations where the generated Go code does not conform to standard Go idioms / conventions / best practices. Taking [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments) as a primary source, and generally trying to follow the standards in the Go std lib and in popular OSS libraries.

For example:
- Use proper case: e.g. a generated object named `ProjectApi` should be `ProjectAPI`.
- The `Configuration` object should be named `Config`.
- Do we need to export `APIClient` at all? What's the use case for having this public?
##### Swagger-codegen version

`master`

**Comment by wing328** [comment]
@neilotoole sounds good to me as long as it conforms to the "Go Code Review Comments"

cc @guohuang

**Comment by vbatts** [comment]
discovered a small bit on swagger-codegen-cli-2.2.3.jar and fixed in https://github.com/swagger-api/swagger-codegen/pull/6669
There is still an issue of "how to populate the `infoEmail` field?" so there is not a trialing whitespace on that  line as well.

But after building with master, it looks like the refactor for 2.3.0 introduced new issues.
A useful tool in checking is `gofmt` or `go get golang.org/x/tools/cmd/goimports`
1) indentation of imports, and placement of non-stdlib paths should be at the end of the import set:
This is the diff of 2.2.3 -> 2.3.0-SNAPSHOT (96444d3)
```diff
@@ -12,22 +12,69 @@ package api
 
 import (
        "bytes"
+       "encoding/json"
+       "encoding/xml"
        "fmt"
-       "io/ioutil"
+       "errors"
+       "io"
+       "mime/multipart"
+    "golang.org/x/oauth2"
+    "golang.org/x/net/context"
+       "net/http"
        "net/url"
+       "time"
+       "os"
        "path/filepath"
        "reflect"
+       "regexp"
        "strings"
+       "unicode/utf8"
+       "strconv"
+)
```
it ought to be like
```go
import (
        "bytes"
        "encoding/json"
        "encoding/xml"
        "errors"
        "fmt"
        "io"
        "mime/multipart"
        "net/http"
        "net/url"
        "os"
        "path/filepath"
        "reflect"
        "regexp"
        "strconv"
        "strings"
        "time"
        "unicode/utf8"

        "golang.org/x/net/context"
        "golang.org/x/oauth2"
)
```

2) alignment: these variable assignment align to the "=" sign. 
```diff
+var (
+       jsonCheck = regexp.MustCompile("(?i:[application|text]/json)")
+       xmlCheck = regexp.MustCompile("(?i:[application|text]/xml)")
 )
```
it ought to be like
```go
        jsonCheck = regexp.MustCompile("(?i:[application|text]/json)")
        xmlCheck  = regexp.MustCompile("(?i:[application|text]/xml)")
```

This is true for fields of a struct as well, they align to the type, as well as the tag on the field. 
Here is a diff of the corrected output from 2.3.0-SNAPSHOT after `goimports -w ...`
```diff
 type Configuration struct {
-       BasePath      string                    `json:"basePath,omitempty"`
-       Host          string                    `json:"host,omitempty"`
-       Scheme        string                    `json:"scheme,omitempty"`
-       DefaultHeader map[string]string         `json:"defaultHeader,omitempty"`
-       UserAgent     string                    `json:"userAgent,omitempty"`
-       HTTPClient        *http.Client
+       BasePath      string            `json:"basePath,omitempty"`
+       Host          string            `json:"host,omitempty"`
+       Scheme        string            `json:"scheme,omitempty"`
+       DefaultHeader map[string]string `json:"defaultHeader,omitempty"`
+       UserAgent     string            `json:"userAgent,omitempty"`
+       HTTPClient    *http.Client
 }
```

3) no space between function name and parenthesis in function signature.
Here is a diff from 2.3.0-SNAPSHOT after `goimports -w ...`:
```diff
 // Change base path to allow switching to mocks
-func (c *APIClient) ChangeBasePath (path string) {
+func (c *APIClient) ChangeBasePath(path string) {
        c.cfg.BasePath = path
```

4) single variable returns in a function do not need parenthesis:
Here is a diff from 2.3.0-SNAPSHOT after `goimports -w ...`:
```diff
 // Prevent trying to import "fmt"
-func reportError(format string, a ...interface{}) (error) {
+func reportError(format string, a ...interface{}) error {
        return fmt.Errorf(format, a...)
```

5) initialized objects with empty curly brackets need no space.
Here is a diff from 2.3.0-SNAPSHOT after `goimports -w ...`:
```diff
        // to determine the Content-Type header
-       localVarHttpContentTypes := []string{  }
+       localVarHttpContentTypes := []string{}
 
        // set Content-Type header
        localVarHttpContentType := selectHeaderContentType(localVarHttpContentTypes)
@@ -69,8 +69,7 @@ func (a *GrafeasApiService) CreateNote(projectsId string, localVarOptionals map[
        }
 
        // to determine the Accept header
-       localVarHttpHeaderAccepts := []string{
-               }
+       localVarHttpHeaderAccepts := []string{}
```

**Comment by wing328** [comment]
@vbatts I've filed #6695 to fix a few formatting issues.
